vertx-web/src/main/java/io/vertx/ext/web/handler/sockjs/impl/BaseTransport.java

```diff
@@ -156,7 +156,11 @@ static void setJSESSIONID(SockJSHandlerOptions options, RoutingContext rc) {
     }
   }
 
-  static void setCORS(RoutingContext rc) {
+  static void setCORS(SockJSHandlerOptions options, RoutingContext rc) {
+    if (!options.isCorsHeadersEnabled()) {
```

Comment:
What I was trying to say here is that, if the CORS handler is used before it can be used to fully control how the header should be presented (or even excluded) and in this case we could just verify if the header is already set instead of having an explicit enable/disable flag. WDYT?